### PR TITLE
Add integration jobs to ansible.utils

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gosriniv@redhat.com>

### CODE REVIEW

**Comment by GomathiselviS** [comment]
recheck

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -421,6 +421,29 @@
     vars:
       ansible_test_python: 3.6
 
+- job:
+    name: ansible-test-network-integration-ansible-utils
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:425]: we should call this ansible-test-integration-ansible-utils

File: zuul.d/jobs.yaml [hunk 2]
```diff
@@ -421,6 +421,29 @@
...
+    abstract: true
+    dependencies:
+      - name: build-ansible-collection
+        soft: true
+    parent: ansible-network-eos-appliance
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:430]: we can drop this, perhaps the parent should be

ansible-test-integration-base, and we can move common bits into it

File: zuul.d/jobs.yaml [hunk 3]
```diff
@@ -678,6 +706,13 @@
     vars:
       ansible_test_python: 3.8
 
+- job:
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:709]: you need to add the controller-nodeset here for python36

**Comment by GomathiselviS** [comment]
recheck

File: zuul.d/jobs.yaml [hunk 4]
```diff
@@ -421,6 +421,34 @@
     vars:
       ansible_test_python: 3.6
 
+- job:
+    name: ansible-test-integration-base
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:425]: we should parent to ansible-test-base, to remove a lot of the duplicate code in our playbooks

File: playbooks/ansible-test-integration-base/run.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (pabelanger) [inline playbooks/ansible-test-integration-base/run.yaml:1]: isn't this duplicate of ansible-test-base/run.yaml?

File: zuul.d/jobs.yaml [hunk 6]
```diff
@@ -678,6 +707,14 @@
     vars:
       ansible_test_python: 3.8
 
+- job:
+    name: ansible-test-network-integration-ansible-utils-python36
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:711]: ansible-test-integration-ansible-utils-python36

File: zuul.d/jobs.yaml [hunk 7]
```diff
@@ -678,6 +707,14 @@
...
+    parent: ansible-test-integration-ansible-utils
+    nodeset: controller-python36
+    vars:
+      ansible_test_collections: true
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:715]: we could move this up into the parent

File: zuul.d/jobs.yaml [hunk 8]
```diff
@@ -421,6 +421,35 @@
...
+
+- job:
+    name: ansible-test-integration-ansible-utils
+    abstract: true
+    parent: ansible-test-integration-base
+    required-projects:
+      - name: github.com/ansible/ansible
+      - name: github.com/ansible-collections/ansible.utils
+    vars:
+      ansible_test_collections: true
```
Comment (pabelanger) [inline zuul.d/jobs.yaml:450]: can drop, duplicate

File: zuul.d/jobs.yaml (hunk 7 above)
Comment (pabelanger) [reply zuul.d/jobs.yaml:715]: actually, duplicate can drop

File: playbooks/ansible-test-integration-base/post.yaml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (pabelanger) [inline playbooks/ansible-test-integration-base/post.yaml:1]: This loops to be duplicate of ansible-test-base/post.yaml, can we use it?

File: zuul.d/project-templates.yaml [hunk 10]
```diff
@@ -99,6 +99,16 @@
       queue: integrated
       jobs: *ansible-collections-arista-eos-jobs
 
+- project-template:
+    name: ansible-collections-ansible-utils
+    check:
+      jobs: &ansible-collections-ansible-utils-jobs
+        - build-ansible-collection
```
Comment (pabelanger) [inline zuul.d/project-templates.yaml:106]: we can remove build-ansible-collecton below in units

File: playbooks/ansible-test-integration-base/pre.yaml [hunk 11]
```diff
@@ -0,0 +1,22 @@
+---
+- hosts: controller
+  tasks:
+    - name: Setup tox role
+      include_role:
+        name: tox
+      vars:
+        tox_envlist: venv
+        tox_install_siblings: false
+        zuul_work_dir: "{{ zuul.projects['github.com/ansible/ansible-zuul-jobs'].src_dir }}"
```
Comment (pabelanger) [inline playbooks/ansible-test-integration-base/pre.yaml:10]: what is the purpose of this venv? What will use it

File: playbooks/ansible-test-integration-base/pre.yaml [hunk 12]
```diff
@@ -0,0 +1,14 @@
+---
+- hosts: controller
+  tasks:
+    - name: Install binary dependencies from ansible/ansible-zuul-jobs
+      include_role:
+        name: bindep
+      vars:
+        bindep_dir: "{{ zuul.projects['github.com/ansible/ansible-zuul-jobs'].src_dir }}"
```
Comment (pabelanger) [inline playbooks/ansible-test-integration-base/pre.yaml:8]: I actually don't think we need this, it was only used in network-integrations to bootstrap appliance dependencies

**Pull request merged**